### DISCUSSION TITLE
Feature: Allow for creating a command in one line (plus import)

### DISCUSSION THREAD

**Discussion by jaraco** [fastapi/typer#928] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

Note, searching for `__name__` or `__main__` in GitHub is ineffective, so a I scanned through issues, but couldn't land on a good search methodology.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import autocommand


@autocommand.autocommand(__name__)
def main(name: str):
	print(f"Hello {name}"))
```

### Description

In [autocommand](https://github.com/Lucretiel/autocommand), the "hello world" is three lines shorter than the same for typer, because it allows a function to be decorated as "the command". Because [autocommand is abandoned](https://github.com/Lucretiel/autocommand/issues/38), I'm considering replacing it with typer. I'd like, however, not to add all of this boilerplate to each of my command scripts. I'd much rather typer provide a decorator that either:

- takes as input the `__name__` parameter and does the `== "__main__"` check, or
- as [proposed here](https://github.com/Lucretiel/autocommand/issues/37), infer the __globals__ from the call stack (similar to what [importlib.resources does](https://github.com/python/cpython/blob/1dad23edbc9db3a13268c1000c8dd428edba29f8/Lib/importlib/resources/_common.py#L90-L104), such that supplying `__name__` is also unnecessary.

Imagine, something like:

```python
import typer

@typer.autorun
def main(name: str):
    typer.echo(f"Hello {name}"))
```

Where `autorun` figures out its call scope, inspects the `f_globals['__name__']` and runs the command if it equals `__main__`.

This one change would go a long way to making typer a suitable replacement for autocommand.

### Operating System

Other

### Typer Version

n/a

### Python Version

n/a

**Comment by johnslavik** [comment]
As in https://github.com/Lucretiel/autocommand/issues/37, we might use `FunctionType.__module__` instead of frame inspection to figure out the moment to auto-run. This however assumes that `typer.autorun` is _always_ used as a decorator (using only the `@` syntax; not `typer.autorun(func)` for instance).
